src/http/httpProxy.ts

```diff
@@ -117,3 +136,37 @@ export async function setupHttpProxy(): Promise<string> {
         });
     });
 }
+
+// Function to get a random port within a specified range
+function getRandomPort(): number {
+    return Math.floor(Math.random() * (maxPort - minPort + 1)) + minPort;
+}
+
+// Function to find an open port starting from a specified port
+function findOpenPort(callback: (port: number) => void): void {
+    const server = net.createServer();
```

Comment:
The server we created is already listening in on port 0. We could use that one, but it would be closing and re-listening a lot. I created a new one to keep the logic cleaner and because using `net.createServer()` should be faster when just checking the ports.